src/main/java/de/hhu/stups/plues/ui/controller/PartialTimeTables.java

```diff
@@ -0,0 +1,265 @@
+package de.hhu.stups.plues.ui.controller;
+
+import com.google.inject.Inject;
+
+import de.hhu.stups.plues.Delayed;
+import de.hhu.stups.plues.data.Store;
+import de.hhu.stups.plues.data.entities.AbstractUnit;
+import de.hhu.stups.plues.data.entities.Course;
+import de.hhu.stups.plues.data.entities.Module;
+import de.hhu.stups.plues.prob.FeasibilityResult;
+import de.hhu.stups.plues.tasks.PdfRenderingTask;
+import de.hhu.stups.plues.tasks.PdfRenderingTaskFactory;
+import de.hhu.stups.plues.tasks.SolverService;
+import de.hhu.stups.plues.tasks.SolverTask;
+import de.hhu.stups.plues.ui.components.CheckBoxGroup;
+import de.hhu.stups.plues.ui.components.CheckBoxGroupFactory;
+import de.hhu.stups.plues.ui.components.MajorMinorCourseSelection;
+
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.ObjectProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.beans.property.SimpleObjectProperty;
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.fxml.Initializable;
+import javafx.scene.Node;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.layout.GridPane;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.VBox;
+import javafx.scene.text.Text;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.ResourceBundle;
+import java.util.concurrent.ExecutorService;
+
+public class PartialTimeTables extends GridPane implements Initializable {
+
+  private final Delayed<Store> delayedStore;
+  private final Delayed<SolverService> delayedSolverService;
+
+  private final BooleanProperty solverProperty;
+  private final BooleanProperty generationStarted;
+  private final BooleanProperty checkStarted;
+  private final CheckBoxGroupFactory checkBoxGroupFactory;
+  private final SimpleObjectProperty storeProperty;
+  private final BooleanProperty resultProperty;
+  private final SimpleBooleanProperty storeAvailable;
+
+  private final PdfRenderingTaskFactory renderingTaskFactory;
+  private final ExecutorService executor;
+  private PdfRenderingTask task;
+  private ObjectProperty<Path> pdf;
+  private Course major;
```

Comment:
Yes, because it needed both in finding modules and units and in creating a pdf out of the selection